meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1634 @@
+        armature = obj.parent
+
+        # Append animation name to armature name
+        armature.name = armature.name + "_" + os.path.basename(self.filepath).replace(".npz", "")
+
+        context.scene.render.fps = target_framerate
+        context.scene.frame_start = 1
+
+        # Set shape and update joint locations
+        # TODO once we have the regressor for SMPLH, we can remove this condition
+        if SMPL_version != 'SMPLH':
+            bpy.ops.object.mode_set(mode='OBJECT')
+            for index, beta in enumerate(betas):
+                key_block_name = f"Shape{index:03}"
+
+                if key_block_name in obj.data.shape_keys.key_blocks:
+                    obj.data.shape_keys.key_blocks[key_block_name].value = beta
+                else:
+                    print(f"ERROR: No key block for: {key_block_name}")
+
+        bpy.ops.object.update_joint_locations('EXEC_DEFAULT')
+
+        # Keyframe poses
+        step_size = int(fps / target_framerate)
+
+        num_frames = trans.shape[0]
+        num_keyframes = int(num_frames / step_size)
+
+        if self.keyframe_corrective_pose_weights:
+            print(f"Adding pose keyframes with keyframed corrective pose weights: {num_keyframes}")
+        else:
+            print(f"Adding pose keyframes: {num_keyframes}")
+
+        if len(bpy.data.actions) == 0:
+            # Set end frame if we don't have any previous animations in the scene
+            context.scene.frame_end = num_keyframes
+        elif num_keyframes > context.scene.frame_end:
+            context.scene.frame_end = num_keyframes
+
+        joint_names = get_joint_names(SMPL_version)
+        joints_to_use = joint_names
+
+        # override hand pose if it's selected
+        # don't pose the hands every frame if we're overriding it
+        if (self.hand_pose != 'disabled'):
+            bpy.ops.object.set_hand_pose('EXEC_DEFAULT')
+
+            if SMPL_version == 'SMPLH':
+                joints_to_use = joint_names[:22]
+            else:
+                joints_to_use = joint_names[:25]
+                print (joints_to_use)
+
+        for index, frame in enumerate(range(0, num_frames, step_size)):
+            if (index % 100) == 0:
+                print(f"  {index}/{num_keyframes}")
+            current_frame = index + 1
+            current_pose = poses[frame].reshape(-1, 3)
+            current_trans = trans[frame]
+
+            for bone_index, bone_name in enumerate(joints_to_use):
+                if bone_name == "pelvis":
+                    # there's a scale mismatch somewhere and the global translation is off by a factor of 100
+                    armature.pose.bones[bone_name].location = Vector((current_trans[0]*100, current_trans[1]*100, current_trans[2]*100))
+                    armature.pose.bones[bone_name].keyframe_insert('location', frame=current_frame)
+
+                # Keyframe bone rotation
+                pose_rodrigues = current_pose[bone_index]
```

Comment:
Making a temp-var just for one use is a little extra, it'd be cleaner to just use `current_pose[bone_index]` in the line below.